src/arch/x86.rs

```diff
@@ -64,13 +66,38 @@ pub unsafe fn init(stack: &Stack, f: unsafe extern "C" fn(usize) -> !) -> StackP
         # of `swap_trampoline` does.
         .cfi_def_cfa %ebx, 0
         .cfi_offset %ebp, -8
-        # Call the next trampoline.
-        call   ${0:c}
+
+        # This nop is here so that the initial swap doesn't return to the start
+        # of the trampoline, which confuses the unwinder since it will look for
+        # frame information in the previous symbol rather than this one. It is
+        # never actually executed.
+        nop
+
+        # Stack unwinding in OSX doesn't seem to like 1-byte symbols, so we add
```

Comment:
s/OSX/libunwind/, since I believe this will affect anyone using libunwind (like ARTIQ)